integreat_compass/cms/views/offers/offer_form_view.py

```diff
@@ -0,0 +1,158 @@
+
+        return render(
+            request,
+            self.template_name,
+            {
+                **self.get_context_data(**kwargs),
+                "offer_form": offer_form,
+                "offer_version_form": offer_version_form,
+                "contact_form": contact_form,
+                "location_form": location_form,
+                "organization_form": organization_form,
+            },
+        )
+
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
+        r"""
+        Submit :class:`~integreat_compass.cms.forms.offers.contact_form.ContactForm`,
+               :class:`~integreat_compass.cms.forms.offers.location_form.LocationForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_form.OfferForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_version_form.OfferVersionForm` and
+               :class:`~integreat_compass.cms.forms.offers.organization_form.OrganizationForm`
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offer_form = OfferForm(
+            data=request.POST, additional_instance_attributes={"creator": request.user}
+        )
+        offer_version_form = OfferVersionForm(data=request.POST)
+        contact_form = ContactForm(
+            data=request.POST, additional_instance_attributes={"creator": request.user}
+        )
+        location_form = LocationForm(data=request.POST)
+        organization_form = OrganizationForm(
+            data=request.POST, additional_instance_attributes={"creator": request.user}
+        )
+
+        forms = (
+            offer_form,
+            offer_version_form,
+            contact_form,
+            location_form,
+            organization_form,
+        )
+        if not all(form.is_valid() for form in forms):
+            for form in forms:
+                form.add_error_messages(request)
+        else:
+            offer_form.instance.offer_contact = contact_form.save()
+            offer_form.instance.location = location_form.save()
+            offer_form.instance.organization = organization_form.save()
+
+            offer_version_form.instance.offer = offer_form.save()
+            offer_version_instance = offer_version_form.save()
+
+            messages.success(
+                request,
+                _(
+                    'Application for the offer "{}" has successfully been submitted and will be reviewed shortly.'
+                ).format(offer_version_instance.title),
+            )
+
```

Comment:
```suggestion
            return redirect(
                "cms:protected:edit_offer", **{"pk": offer_form.instance.id}
            )

```
Typically, you want to redirect after a successful form submission, otherwise the same data might get submitted again when the page is refreshed...

Reply:
Yes, though I think I'd rather redirect to the user's offer overview, but since this did not exist yet, the placeholder.